### PR TITLE
Minute level partitions

### PR DESCRIPTION
**Pull request by lovenishgoyal**
Minute level partition to push data on s3

### CODE REVIEW

File: src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java [hunk 1]
```diff
@@ -121,49 +146,63 @@ private long getFinalizedTimestampMillis(Message lastMessage,
 
     @Override
     public String[] getFinalizedUptoPartitions(List<Message> lastMessages,
-                                               List<Message> committedMessages) throws Exception {
+                                               List<Message> committedMessages) throws Exception
+    {
+
         if (lastMessages == null || committedMessages == null) {
-            LOG.error("Either: {} and {} is null", lastMessages,
-                committedMessages);
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java]: Why this line is removed?

File: src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java [hunk 2]
```diff
@@ -121,49 +146,63 @@ private long getFinalizedTimestampMillis(Message lastMessage,
...
         }
         assert lastMessages.size() == committedMessages.size();
 
         long minMillis = Long.MAX_VALUE;
         for (int i = 0; i < lastMessages.size(); i++) {
-            long millis = getFinalizedTimestampMillis(lastMessages.get(i),
-                committedMessages.get(i));
+            long millis = getFinalizedTimestampMillis(lastMessages.get(i), committedMessages.get(i));
             if (millis < minMillis) {
-                LOG.info("partition {}, time {}", i, millis);
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java]: Why this line is removed?

File: src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java [hunk 3]
```diff
@@ -121,49 +146,63 @@ private long getFinalizedTimestampMillis(Message lastMessage,
...
         }
         if (minMillis == Long.MAX_VALUE) {
-            LOG.error("No valid timestamps among messages: {} and {}", lastMessages,
-                committedMessages);
+            LOG.error("No valid timestamps among messages: {} and {}", lastMessages, committedMessages);
             return null;
         }
 
         // add the safety lag for late-arrival messages
-        minMillis -= mFinalizerDelaySeconds * 1000L;
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java]: Why this line is changed?

File: src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java [hunk 4]
```diff
@@ -121,49 +146,63 @@ private long getFinalizedTimestampMillis(Message lastMessage,
...
 
     @Override
-    public String[] getPreviousPartitions(String[] partitions) throws Exception {
+    public String[] getPreviousPartitions(String[] partitions) throws Exception
+    {
+
         long millis = parsePartitions(partitions);
         boolean usingHourly = mUsingHourly;
-        if (mUsingHourly && millis % DAY_IN_MILLIS == 0) {
-            // On the day boundary, if the currrent partition is [dt=07-07, hr=00], the previous
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java]: Why the comment is removed?

**Comment by HenryCaiHaiying** [comment]
On the high level, if you use minute folder, you will generate lots of small files on S3 (and a lot of hive partitions), this would affect your downstream jobs performance if they need to a directory listing (which is common for Hadoop InputFormat classes)

Also can you squash three commits into one?

**Comment by lovenishgoyal** [comment]
Hi Henry,

Thanks for review the changes. I have gone through your comments, I have
analysed and found it was just a code format rather than code change. Also
as you mentioned "small files on S3" concern it is a feature that provide
data on s3 on minute level partition. We can use combiner to deal with such
small files.

I will merge all three commits and raise a new pull request for same.

Thanks,
Lovenish

On Fri, Mar 18, 2016 at 10:50 PM, Henry Haiying Cai <
notifications@github.com> wrote:

> On the high level, if you use minute folder, you will generate lots of
> small files on S3 (and a lot of hive partitions), this would affect your
> downstream jobs performance if they need to a directory listing (which is
> common for Hadoop InputFormat classes)
>
> Also can you squash three commits into one?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pinterest/secor/pull/190#issuecomment-198460307>
>



-- 
*Regards,*
Lovenish Goyal
*M: +91 9545461923** | Skype:  lovenishgoyal*

**Pull request closed without merging**